components/SelectInventoriedResource.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import {gql, useQuery} from "@apollo/client";
+import {useAuth} from "../lib/auth";
+
+type Resource = { id: string, name: string, onhandQuantity: any }
+type handleResource = (value: Resource) => void
+
+
+const SelectInventoriedResource: any = (props: { handleSelect: handleResource }) => {
+    const {authId} = useAuth()
+    const FETCH_INVENTORY = gql(`query($id: ID!) {
+                                      agent(id: $id) {
+                                        inventoriedEconomicResources {
+                                          id
+                                          name
+                                          onhandQuantity {
+                                            id
+                                            hasNumericalValue
+                                            hasUnit {
+                                              label
+                                              symbol
+                                            }
+                                          }
+                                        }
+                                      }
+                                    }`)
+
+
+    const resources: Array<Resource> = useQuery(FETCH_INVENTORY,{variables: {id: authId}}).data?.agent.inventoriedEconomicResources
```

Comment:
In this very case, where FETCH_INVENTORY is used once... and the variables are passed into it I prefer to read:

```js
const resources: Array<Resource> = useQuery(gql`......balbalba....`, { variables: { id: authId }}).data?.agent.inventoriedEconomicResources
```

and again pay attention to linting